docs/manual/common/concepts/ServiceDiscovery.md

```diff
@@ -1,33 +1,51 @@
+A Service Registry collaborates with microservice instances to maintain an up-to-date lookup table. The table includes the host and port for each available microservice instance. As load changes, the system can spawn or destroy instances in any location while continuing to satisfy requests. You can design a system to enable microservices to self-register or you can use a [3rd Party Registration service](http://microservices.io/patterns/3rd-party-registration.html).
+
+When booting a Lagom microservice instance, a registrar will register the name of the microservice, the URL and the names of the locatable Service Descriptors on the Service Registry so that they can be located. When powering down an instance of a service, the registrar will have to update the Service Registry too. Lagom's [[Developer Environment|DevEnvironment]] provides an implementation of a Service Registry and a registrar so you can run your microservices locally.
+ 
+<!---The following diagram illustrates service registration. (TBA) --->
+
+Many available technologies provide Service Registry capabilities. You will need to choose and/or develop a Service Locator for your services to run in your deployment environment (see for example [Lagom ZooKeeper Service Locator](https://github.com/jboner/lagom-service-locator-zookeeper)). You may need to work out a way to plug your Lagom services with a registrar. Lagom's integration with [[ConductR|ConductR]] makes these two steps seamless.
+
+
+## Client-side service discovery
 
 From Bonér's [Reactive Microservices Architecture: Design Principles for Distributed Systems](http://www.oreilly.com/programming/free/reactive-microservices-architecture.html) 
 
 > Once the information about each service has been stored it can be made available through a Service Registry that services can use to look the information up—using a pattern called Client-Side Service Discovery.
 
-Lagom creates service clients for each Service Descriptor so that applications can interact with Lagom Services. So [[an app|IntegratingNonLagom]] that wanted to consume the hello service can use the Welcome Service Client and simply invoke the `hello` method. The Welcome Service Client will use the Service Registry to find a valid URL where  `welcome` is available and with that information the request is then fulfilled. This approach requires using Lagom provided code.
+Lagom creates service clients for each Service Descriptor so that applications can interact with Lagom Services. Suppose a non-Lagom [[an app|IntegratingNonLagom]] app wants to consume a hello service. It can use the Welcome Service Client and simply invoke the `hello` method. The Welcome Service Client will use the Service Registry to find a valid URL where  `welcome` is available and fulfill the request. This approach requires using Lagom provided code. In production, the Service Locator plugged into your services will be an element participating in this Client-Side Discovery.
```

Comment:
"Suppose a non-Lagom [[an app|IntegratingNonLagom]] app wants to..." I think that this would render as: 

 * `Suppose a non-Lagom an app app wants to...`

I think this is not the expected outcome.